src/compressed_tensors/logger.py

```diff
@@ -0,0 +1,132 @@
+    log_file: Optional[str] = None
+    log_file_level: Optional[str] = None
+
+
+def configure_logger(config: Optional[LoggerConfig] = None) -> None:
+    """
+    Configure the logger for Compressed Tensors.
+    This function sets up the console and file logging
+    as per the specified or default parameters.
+
+    Note: Environment variables take precedence over the function parameters.
+
+    :param config: The configuration for the logger to use.
+    :type config: LoggerConfig
+    """
+    logger_config = config or LoggerConfig()
+
+    # env vars get priority
+    if (disabled := os.getenv("COMPRESSED_TENSORS_LOG_DISABLED")) is not None:
+        logger_config.disabled = disabled.lower() == "true"
+    if (clear_loggers := os.getenv("COMPRESSED_TENSORS_CLEAR_LOGGERS")) is not None:
+        logger_config.clear_loggers = clear_loggers.lower() == "true"
+    if (console_log_level := os.getenv("COMPRESSED_TENSORS_LOG_LEVEL")) is not None:
+        logger_config.console_log_level = console_log_level.upper()
+    if (log_file := os.getenv("COMPRESSED_TENSORS_LOG_FILE")) is not None:
+        logger_config.log_file = log_file
+    if (log_file_level := os.getenv("COMPRESSED_TENSORS_LOG_FILE_LEVEL")) is not None:
+        logger_config.log_file_level = log_file_level.upper()
+
+    if logger_config.disabled:
+        logger.disable("compressed_tensors")
+        return
+
+    logger.enable("compressed_tensors")
+
+    if logger_config.clear_loggers:
+        logger.remove()
+
+    if logger_config.console_log_level:
+        # log as a human readable string with the time, function, level, and message
+        logger.add(
+            sys.stdout,
+            level=logger_config.console_log_level.upper(),
+            format="{time} | {function} | {level} - {message}",
+            filter=support_log_once,
+        )
+
+    if logger_config.log_file or logger_config.log_file_level:
+        log_file = logger_config.log_file or "compressed_tensors.log"
+        log_file_level = logger_config.log_file_level or "INFO"
+        # log as json to the file for easier parsing
+        logger.add(
+            log_file,
+            level=log_file_level.upper(),
+            serialize=True,
+            filter=support_log_once,
+        )
+
+
+def support_log_once(record: Dict[str, Any]) -> bool:
+    """
+    Support logging only once using `.bind(log_once=True)`
+
+    ```
+    logger.bind(log_once=False).info("This will log multiple times")
+    logger.bind(log_once=False).info("This will log multiple times")
+    logger.bind(log_once=True).info("This will only log once")
+    logger.bind(log_once=True).info("This will only log once")  # skipped
+    ```
+    """
+    log_once = record["extra"].get("log_once", False)
+    level = getattr(record["level"], "name", "none")
+    message = str(level) + record["message"]
+
+    if log_once and message in _logged_once:
+        return False
+
+    if log_once:
+        _logged_once.add(message)
```

Comment:
I am concerned that this could be a bit of a memory leak in long running processes. If we are printing a lot this set will grow (and things like tensors could be quite large when printing).

If we want to use this approach we should at least use `hash(message)` instead of `message`.

Reply:
This will only append once for each type of log we have, so i don't think memory leak is a concern. hashing might be a good idea, but they don't reproduce across different processes or subprocesses. and from what i've read in a quick search it does not for multiprocessing either

Reply:
Hashing is a good idea, but I agree that this is not really a concern.

Different processes have different memory. Also strings are always hashed consistently.